kitsune/community/utils.py

```diff
@@ -1,75 +1,116 @@
 from datetime import datetime, date, timedelta
+from django.core.cache import cache
 from django.conf import settings
+from django.db.models import Count
 
 from kitsune.customercare.models import ReplyMetricsMappingType
 from kitsune.products.models import Product
 from kitsune.questions.models import AnswerMetricsMappingType
 from kitsune.search.es_utils import F
-from kitsune.users.models import UserMappingType
+from kitsune.users.models import User
+from kitsune.users.templatetags.jinja_helpers import profile_avatar
 from kitsune.wiki.models import RevisionMetricsMappingType
 
 
 # This should be the higher than the max number of contributors for a
 # section.  There isn't a way to tell ES to just return everything.
 BIG_NUMBER = 3000
 
+# This should be higher than the max number of questions, revisions or tweets
+HUGE_NUMBER = 1500000
+
 
 def top_contributors_questions(start=None, end=None, locale=None, product=None,
-                               count=10, page=1):
+                               count=10, page=1, use_cache=True):
     """Get the top Support Forum contributors."""
     # Get the user ids and contribution count of the top contributors.
-    query = (
-        AnswerMetricsMappingType
-        .search()
-        .facet('creator_id', filtered=True, size=BIG_NUMBER))
+
+    cache_key = 'top_contributors_questions_{}_{}_{}_{}_{}_{}'.format(start, end, locale,
+                                                                      product, count, page)
+    cached = cache.get(cache_key, None)
+    if use_cache and cached:
+        return cached
+
+    query = AnswerMetricsMappingType.search()
 
     # Adding answer to your own question, isn't a contribution.
     query = query.filter(by_asker=False)
-
     query = _apply_filters(query, start, end, locale, product)
 
-    return _get_creator_counts(query, count, page)
+    answers = [q.id for q in query.all()[:HUGE_NUMBER]]
```

Comment:
I think it can be like this. More faster to query I believe!
```python
users_id = [q.creator_id for q in query.all()[:HUGE_NUMBER]]
user = User.objects.filter(id__in=set(users_id)).annotate(query_count=Count('answers')).order_by('-query_count')
```